### PR TITLE
initial Spring Boot 2.0.x refactoring

### PR DESCRIPTION
**Pull request by Salaboy**
First round to update parent dependencies to Spring Boot 2.0.0.BUILD-SNAPSHOT. 

https://github.com/spring-cloud-incubator/spring-cloud-kubernetes/issues/139

I will start simple by disabling some modules. I want to make sure that I can cover 
- Spring Cloud Kubernetes :: Dependencies 
- Spring Cloud Kubernetes 
- Spring Cloud Kubernetes :: Core 
- Spring Cloud Kubernetes :: Config
- Spring Cloud Kubernetes :: Discovery

and related examples first before jumping into the Netflix related projects. 
Help is appreciated. 
I'm sharing this in a PR so if people want to start working on other modules they can.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@Salaboy Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud-incubator/spring-cloud-kubernetes&pullRequestId=152)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud-incubator/spring-cloud-kubernetes&pullRequestId=152) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@Salaboy Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud-incubator/spring-cloud-kubernetes&pullRequestId=152)!

**Comment by Salaboy** [comment]
@spencergibb the tests for config are now failing due this commit: https://github.com/spring-projects/spring-boot/commit/1d83e87bcdcd0a70680ac834900ddd3821cc65b9#diff-77ef491ee95a5312174e8ecd21964d10 there seems to be a project that hasn't been updated to work against that commit in Spring Cloud.

**Comment by Salaboy** [comment]
https://github.com/spring-cloud/spring-cloud-commons/issues/324
Already reported and I think that is going to be fixed soon.

**Comment by Salaboy** [comment]
This is compiling now, but I will need to do more testing to check that it actually works in kubernetes.

**Comment by Salaboy** [comment]
- Discovery working after my last commit
now time to work on ConfigMaps

**Comment by Salaboy** [comment]
I'm planning to see if I can fix this one as well, which is needed for the fabric8 plugin to work with spring book 2 -> https://github.com/fabric8io/fabric8-maven-plugin/issues/1206

**Comment by Salaboy** [comment]
@spencergibb can you please check this? both examples (discovery and configmaps) are working. 
I think that this is a good base for moving this forward in the master branch.. Is it possible to merge so I can start consuming the snapshots .. I'm pretty sure that when I start using this in real life projects I will find a lot of improvements.

File: pom.xml [hunk 1]
```diff
@@ -59,36 +59,36 @@
 
 	<properties>
 		<!-- Dependency Versions -->
-		<spring-cloud-commons.version>1.2.3.RELEASE</spring-cloud-commons.version>
-		<spring-cloud-netflix.version>1.3.2.RELEASE</spring-cloud-netflix.version>
-		<spring-cloud-sleuth.version>1.2.2.RELEASE</spring-cloud-sleuth.version>
+		<spring-cloud-commons.version>2.0.0.BUILD-SNAPSHOT</spring-cloud-commons.version>
+		<!--<spring-cloud-netflix.version>1.3.2.RELEASE</spring-cloud-netflix.version>-->
```
Comment (spencergibb) [inline pom.xml]: Seems like a big omission.

Comment (Salaboy) [reply pom.xml]: We need to start somewhere, migrating all the projects will take some more time, and I can see people wanting to start with Discovery and ConfigMaps, those two bring value on their own. 
Can we consider this as the initial work?

Comment (Salaboy) [reply pom.xml]: Each of the other module can be migrated one by one, by someone that is interested in

Comment (spencergibb) [reply pom.xml]: That's fine, but it looks like you deleted ribbon classes.

Comment (Salaboy) [reply pom.xml]: Yes that's part of Ribbon of netflix, we can port those classes back from the 1.5 branch when we move the netflix integration

Comment (spencergibb) [reply pom.xml]: I don't think they need to be deleted since the module won't be compiled.

Comment (Salaboy) [reply pom.xml]: @spencergibb sorry to be a pain but can you point me to the classes that you are talking about? I might be missing something big. 

I've removed the modules as well. I don't want to have modules that are not being compiled in the repository, I would rather have them in the original branch and do a progressive migration. That gives us an opportunity to review the modules/dependencies/code in detail.

**Comment by Salaboy** [comment]
Guys, I have updated the examples and READMEs. I've also created an independent project here: https://github.com/Salaboy/minions/tree/spring-cloud-k8s
That depends on this PR. I will add the config maps into that example.

**Comment by shahbour** [comment]
Any plan or timeline on when this request can be merged ?

File: spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesDiscoveryClientConfig.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import org.apache.commons.logging.LogFactory;
+import org.springframework.cloud.client.discovery.DiscoveryClient;
+import org.springframework.cloud.kubernetes.registry.KubernetesRegistration;
+import org.springframework.cloud.kubernetes.registry.KubernetesServiceRegistry;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+
+@Configuration
+public class KubernetesDiscoveryClientConfig {
```
Comment (spencergibb) [inline spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesDiscoveryClientConfig.java]: This is not the standard naming convention of `@Configuration` classes.

File: pom.xml [hunk 3]
```diff
@@ -34,7 +34,7 @@
 	<name>Spring Cloud Kubernetes</name>
 
 	<url>http://cloud.spring.io</url>
-	<inceptionYear>2016</inceptionYear>
+	<inceptionYear>2018</inceptionYear>
```
Comment (spencergibb) [inline pom.xml]: Should be 2017

File: spring-cloud-kubernetes-dependencies/pom.xml [hunk 4]
```diff
@@ -44,36 +42,12 @@
 				<version>${project.version}</version>
 			</dependency>
 
-			<dependency>
-				<groupId>org.springframework.cloud</groupId>
-				<artifactId>spring-cloud-kubernetes-archaius</artifactId>
```
Comment (spencergibb) [inline spring-cloud-kubernetes-dependencies/pom.xml]: Add modules back and comment them out.

File: spring-cloud-kubernetes-discovery/src/main/resources/META-INF/spring.factories [hunk 5]
```diff
@@ -1,3 +1,5 @@
+org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
+org.springframework.cloud.kubernetes.discovery.KubernetesDiscoveryClientConfig
```
Comment (spencergibb) [inline spring-cloud-kubernetes-discovery/src/main/resources/META-INF/spring.factories]: If this is autoconfiguration, it should be named as such.

**Comment by Salaboy** [comment]
@shahbour i will work on spencer comments today so we get it done!

**Comment by Salaboy** [comment]
@spencergibb minor changes fixed.. sorry for not spotting those before.

File: spring-cloud-kubernetes-dependencies/pom.xml (hunk 4 above)
Comment (Salaboy) [reply spring-cloud-kubernetes-dependencies/pom.xml]: This is already down there commented out in line 51, I've moved all the commented modules together. Is there anything else needed regarding this one?

**Pull request merged**